tests/testthat/test-predict.R

```diff
@@ -12,6 +12,12 @@ swiss <-
 part <- caret::createDataPartition(swiss$Catholic, .8)[[1]]
 tr <- swiss[part, ]
 te <- swiss[-part, ]
+te_newlevel <- te
```

Comment:
These bits of code should be functions for reusability, clarity, and readability, and determinism. Having them in a file means that tests could interact with each other. Something like `make_missing_data()` or `make_missing_dataframe()`, which should be called every time an individual test runs.

Reply:
I like the idea of a having testing utility functions like `add_missingness(df, column)`. Going a step further, I like the idea of having a test-aaa.R file that creates one or two data frames that all the tests can be run against. We could do `prep_data`, `tune_models`, etc. there instead of across many test files and drop the time to run tests substantially. Added a note to #826.

Not sure I get the concern about tests interacting. We typically do setup at the top of test files for stuff that will be used across tests. Each `test_that` call is a self-contained environment.... they can read from the parent environment but can't (easily) modify objects there.